platform-sdk/swirlds-base/src/testFixtures/java/com/swirlds/base/test/fixtures/concurrent/WithTestExecutor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base.test.fixtures.concurrent;
+
+import com.swirlds.base.test.fixtures.concurrent.internal.TestExecutorExtension;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * This annotation is used to inject a {@link TestExecutor} instance into a test.
+ *
+ * @see TestExecutor
+ */
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+@ExtendWith(TestExecutorExtension.class)
+public @interface WithTestExecutor {}
```

Comment:
We need to add the new extensions to https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/docs/base/test-support/test-support.md - I created https://github.com/hashgraph/hedera-services/issues/9852